x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/DateTimeFormatProcessor.java

```diff
@@ -36,18 +36,61 @@
         {"F", "S"},
         {"z", "X"}
     };
+
+    private static final String[][] JAVA_TIME_FORMAT_REPLACEMENTS_FOR_MYSQL = {
+        {"%a", "EEE"},
+        {"%b", "MMM"},
+        {"%c", "MM"},
+        {"%D", "d"},
+        {"%d", "dd"},
+        {"%e", "dd"},
+        {"%f", "SSSSSS"},
+        {"%H", "HH"},
+        {"%h", "hh"},
+        {"%I", "hh"},
+        {"%i", "mm"},
+        {"%j", "DDD"},
+        {"%k", "H"},
+        {"%l", "hh"},
+        {"%M", "MMMM"},
+        {"%m", "MM"},
+        {"%p", "a"},
+        {"%r", "hh:mm:ss a"},
+        {"%S", "ss"},
+        {"%s", "ss"},
+        {"%T", "HH:mm:ss"},
+        {"%U", "w"},
+        {"%u", "w"},
+        {"%V", "w"},
+        {"%v", "w"},
+        {"%W", "EEEE"},
+        {"%w", "e"},
+        {"%X", "Y"},
+        {"%x", "w"},
+        {"%Y", "yyyy"},
+        {"%y", "yy"}
+    };
+
     private final Formatter formatter;
 
 
     public enum Formatter {
         FORMAT,
-        DATE_TIME_FORMAT;
+        DATE_TIME_FORMAT,
+        DATE_FORMAT;
 
         private String getJavaPattern(String pattern) {
             if (this == FORMAT) {
-                for (String[] replacement : JAVA_TIME_FORMAT_REPLACEMENTS) {
-                    pattern = pattern.replace(replacement[0], replacement[1]);
-                }
+                pattern = replacePattern(pattern, JAVA_TIME_FORMAT_REPLACEMENTS);
+            } else if (this == DATE_FORMAT) {
+                pattern = replacePattern(pattern, JAVA_TIME_FORMAT_REPLACEMENTS_FOR_MYSQL);
```

Comment:
This is unfortunately not enough. The reason being that the result of `getJavaPattern()` is later fed directly to `DateTimeFormatter.ofPattern()`, which will interpret any other characters which aren't `DATE_FORMAT` specific, but which are `DateTimeFormatter` specific.

`SELECT DATE_FORMAT('2020-03-04T05:06:07Z'::DATETIME, '%A %c')` will not produce `%A 3` (once `c` is corrected), but `%18367000 3` (well, depending on the momentary `milli-of-day`).

This is a current defect with `FORMAT` [too](https://github.com/elastic/elasticsearch/issues/66560). You don't have to necessarily fix `FORMAT`, but a correct implementation of this function would still be needed.